### PR TITLE
cross-repository compatibility check in spring-cloud-gcp

### PR DESCRIPTION
**Pull request by suztomo**
This pull request creates a GitHub Actions workflow that we can trigger when running compatibility checks on Cloud Java client libraries.

Example usage after installation:

```
~/spring-cloud-gcp $ gh workflow run \
    --repo suztomo/spring-cloud-gcp "Compatibility Check" \
    -f mavenRepository="https://google.oss.sonatype.org/content/repositories/snapshots" \
    -f mavenDependencies='{
        "com.google.cloud:google-cloud-bigquery": "2.18.3-SNAPSHOT",
        "com.google.cloud:google-cloud-storage":"2.14.1-SNAPSHOT"
    }'
```

At compatibility check runtime, `force_depencencies.py` replaces the marker "`<!-- {x-cross-repo-check-dependency-mark} -->`" embed in pom.xml with actual forced dependencies with "-SNAPSHOT". 

image

The new "compatibility-check" profile specifies the repository that holds the snapshot version of the library. nhttps://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1319/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R696

The profile is activated only when we specify `compatibilityCheckRepository` property. The `force_dependencies.py` script uses the property.

In my testing with suztomo/spring-cloud-gcp repository clone, the job successfully downloaded the snapshot artifact from the specified repository:

Screen Shot 2022-11-04 at 4 17 57 PM

### CODE REVIEW

**Comment by suztomo** [comment]
@elefeint It's ready for review.

"SonarCloud Code Analysis" failed with "50.0% Coverage on New Code (is less than 80%)". This change does not affect Java code. Do I have to take any actions?

File: .github/workflows/compatibilityCheck.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+jobs:
+  integrationTests:
+    name: Integration Tests for Compatibility Check
+    if: github.repository == 'suztomo/spring-cloud-gcp'
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        it:
+          - bigquery
```
Comment (elefeint) [inline .github/workflows/compatibilityCheck.yaml:27]: Is it intentionally only for 3 products for now?

**Review by elefeint — APPROVED** [review]

File: .github/workflows/compatibilityCheck.yaml (hunk 1 above)
Comment (suztomo) [reply .github/workflows/compatibilityCheck.yaml:27]: I just brought the rest back.

**Comment by suztomo** [comment]
@elefeint Thank you for review and offer to skip SonarCloud. Yes, please merge this.

**Pull request merged**